ccm_web/server/src/api/api.service.ts

```diff
@@ -137,44 +132,62 @@ export class APIService {
     return await sectionHandler.enrollUsers(sectionUsers)
   }
 
-  async enrollSectionExternalUsers (user: User, session: SessionData, sectionId: number, sectionUsers: SectionExternalUserDto[]): Promise<ExternalEnrollments|APIErrorData> {
+  async enrollSectionExternalUsers (user: User, sectionId: number, sectionUsers: SectionExternalUserDto[]): Promise<ExternalEnrollmentResult> {
     // Get requestor/handler and account ID for admin operations
     const adminRequestor = this.canvasService.createRequestorForAdmin('/api/v1/')
     const adminHandler = new AdminApiHandler(adminRequestor, user.loginId)
-    const newUserAccountID =  this.configService.get('canvas.newUserAccountID', { infer: true }) as number
+    const newUserAccountID = this.configService.get('canvas.newUserAccountID', { infer: true })
+
+    const resultData: ExternalEnrollmentUserData = {}
 
     // Create all requested users, noting failures
+    const createErrors: APIErrorData[] = []
+    const newUsers: CanvasUserLoginEmail[] = []
     const createUserResponses = await adminHandler.createExternalUsers(sectionUsers, newUserAccountID)
-    const newUsers = createUserResponses.filter(response => !hasKeys(response, ['error'])) as CanvasUserLoginEmail[]
-
-    console.log('createUserResponses…')
-    console.log(JSON.stringify(createUserResponses))
+    createUserResponses.forEach(({ email, result }) => {
+      let userCreated: false | APIErrorData | CanvasUserLoginEmail
+      if (isAPIErrorData(result)) {
+        if (result.statusCode === HttpStatus.BAD_REQUEST) {
+          userCreated = false
+        } else {
+          userCreated = result
+          createErrors.push(result)
+        }
+      } else {
+        userCreated = result
+        newUsers.push(result)
+      }
+      resultData[email] = {}
+      resultData[email].userCreated = userCreated
```

Comment:
I'm not certain of the syntax, but couldn't this be… 
```suggestion
      resultData[email] = {userCreated: userCreated}
```

Reply:
Yeah, that should work, or even `resultData[email] = { userCreated }`

Reply:
Done.